contracts/Aavegotchi/facets/ItemsRolesRegistryFacet.sol

```diff
@@ -0,0 +1,251 @@
+        );
+    }
+
+    function revokeRoleFrom(uint256 _depositId, bytes32 _role, address _grantee) external override validRoleAndGrantee(_role, _grantee, _depositId) {
+        RoleData memory _roleData = s.itemsRoleAssignments[_depositId];
+        DepositInfo memory _depositInfo = s.itemsDeposits[_depositId];
+
+        address caller = _findCaller(_roleData, _depositInfo);
+        if (_roleData.expirationDate > block.timestamp && !_roleData.revocable) {
+            // if role is not expired and is not revocable, only the grantee can revoke it
+            require(caller == _roleData.grantee, "ItemsRolesRegistryFacet: depositId is not expired or is not revocable");
+        }
+
+        uint256 _gotchiId = s.depositIdToItemIdToGotchiId[_depositId][_depositInfo.tokenId];
+        EquipedDelegatedItemInfo memory _equipedDelegatedItemInfo = s.gotchiIdToItemIdToDepositId[_gotchiId][_depositInfo.tokenId];
+        uint256 _balanceToUnequip = _equipedDelegatedItemInfo.balance;
+
+        if (_balanceToUnequip > 0) {
+            _unequipWearable(_gotchiId, _depositInfo.tokenId, _balanceToUnequip);
+
+            delete s.depositIdToItemIdToGotchiId[_depositId][_depositInfo.tokenId];
+            delete s.gotchiIdToItemIdToDepositId[_gotchiId][_depositInfo.tokenId];
+        }
+
+        delete s.itemsRoleAssignments[_depositId];
+
+        emit RoleRevoked(
+            _depositId,
+            UNIQUE_ROLE,
+            _depositInfo.tokenAddress,
+            _depositInfo.tokenId,
+            _depositInfo.tokenAmount,
+            _depositInfo.grantor,
+            _roleData.grantee
+        );
+    }
+
+    function _unequipWearable(uint256 _gotchiId, uint256 _wearableToUnequip, uint256 _balanceToUnequip) internal {
+        uint256 _unequipedBalance;
+        for (uint256 slot; slot < EQUIPPED_WEARABLE_SLOTS; slot++) {
+            if (_unequipedBalance == _balanceToUnequip) break;
+            if (s.aavegotchis[_gotchiId].equippedWearables[slot] != _wearableToUnequip) continue;
+            s.aavegotchis[_gotchiId].equippedWearables[slot] = 0;
+            _unequipedBalance++;
+        }
+
+        LibItems.removeFromParent(address(this), _gotchiId, _wearableToUnequip, _unequipedBalance);
+        emit LibERC1155.TransferFromParent(address(this), _gotchiId, _wearableToUnequip, _unequipedBalance);
+    }
+
+    function withdraw(uint256 _depositId) public onlyOwnerOrApproved(s.itemsDeposits[_depositId].grantor, s.itemsDeposits[_depositId].tokenAddress) {
+        DepositInfo memory _depositInfo = s.itemsDeposits[_depositId];
+        require(
+            s.itemsRoleAssignments[_depositId].expirationDate < block.timestamp || s.itemsRoleAssignments[_depositId].revocable,
+            "ItemsRolesRegistryFacet: token has an active role"
+        );
+
+        delete s.itemsDeposits[_depositId];
+
+        _transferFrom(address(this), _depositInfo.grantor, _depositInfo.tokenAddress, _depositInfo.tokenId, _depositInfo.tokenAmount);
```

Comment:
Verify if the wearable is equipped before withdrawing. If equipped, call `_unequipDelegatedWearable`